libs/rtemodel/include/RteGenerator.h

```diff
@@ -43,6 +43,12 @@ class RteGenerator : public RteItem
   */
   virtual ~RteGenerator() override;
 
+  /**
+   * @brief get generator commands for all host types without expansion
+   * @return map with host type ("win", "linux", "mac", "other", or "all") as key and generator command as value
+  */
+  std::map<std::string, std::string> GetCommands() const;
```

Comment:
The same as for GetExecutables() : what is the purpose of this method?

Reply:
Rather than implementing the same functionality (taking an `RteItem` object and extracting the list of commands) in both `RteGenerator` and `ProjMgrYamlCbuild`, I thought it would make sense for a generator to return it's commands (and the now removed arguments).

Reply:
I wonder why do you need commands for all platforms, say running on Linux?

Reply:
Because the lockfile we generate shouldn't be different based on which platform it's generated. It should contain the commands for all platforms.

Reply:
But expanded paths are absolute, how can it work for different platforms?

Reply:
By using `ProjMgrYamlCbuild::FormatPath()` on the executable in `ProjMgrYamlCbuild::SetGeneratorsNode()`.